x-pack/plugins/canvas/public/components/workpad/workpad.js

```diff
@@ -104,14 +104,24 @@ export class Workpad extends React.PureComponent {
 
           <Fullscreen>
             {({ isFullscreen, windowSize }) => {
-              const scale = Math.min(windowSize.height / height, windowSize.width / width);
+              const hasHeaderBanner = document.getElementsByClassName('kbnBody--hasHeaderBanner')
+                .length;
+
+              const headerBannerOffset = hasHeaderBanner ? HEADER_BANNER_HEIGHT : 0;
```

Comment:
Sorry, when you asked me how to check if a banner was present in javascript, I thought it was for testing, not for actually using it in a plugin's code.

The proper way with Kibana API would be to add a new `hasHeaderBanner` API, and have the plugin leverage this info.

https://github.com/elastic/kibana/blob/e88d77b0454e26d888c862c3c45a57cfe1836d69/src/core/public/chrome/chrome_service.tsx#L272-L274

Something like 

```ts
      hasHeaderBanner$: () => {
        return headerBanner$.pipe(map((banner) => banner !== undefined))
      },
```

How urgent is this? I could have a PR ready tomorrow to add the API to `coreStart.chrome`.   Not sure how hard to would be to cascade this info from `canvas` mountpoint to this component?

Reply:
Thanks for the help! Not super urgent in my opinion. Just let me know what I'd need to do on my end when you're ready.

Reply:
I created https://github.com/elastic/kibana/pull/111248, I'll ping you once it's merged.

Reply:
https://github.com/elastic/kibana/pull/111248 has been merged and backported to `7.16` and `7.15`. You should now be able to use the `coreStart.chrome.hasHeaderBanner$` API.

Now, about how to retrieve this API down in the `Workpad` component. I took a quick look, but given the multiple layers of composition of the component (e.g `x-pack/plugins/canvas/public/components/workpad/index.js`) and the fact that it's still plain javascript, I'm not comfortable performing the  change myself in your PR, and think that maybe the owner team should be the one to do it.

I see the `canvas` app is wrapping their root tree with the `Kibana` context provider

https://github.com/elastic/kibana/blob/303526ffdbab8c5957a88a7c0a13f44ef9cb2cc3/x-pack/plugins/canvas/public/application.tsx#L73-L74

So in theory the `chrome` core service should be accessible at any depth, using either the `useKibana` hook or the `withKibana` HOC.

So, it may be as easy as using the HOC in `x-pack/plugins/canvas/public/components/workpad/workpad.js`, however I'm not sure about the best practices, like maybe the team prefers to add this wrapper in the composition performed in `x-pack/plugins/canvas/public/components/workpad/index.js`, or even from elsewhere. 

cc @elastic/kibana-presentation I think we need your help here.